core/shared/src/main/scala/zio/Queue.scala

```diff
@@ -150,7 +162,7 @@ object Queue {
             _ <- (IO.effectTotal {
                   unsafeOffer(as, p)
                   unsafeOnQueueEmptySpace(queue)
-                } *> p.await).onInterrupt(IO.effectTotal(unsafeRemove(p)))
+                } *> checkShutdownState *> p.await).onInterrupt(IO.effectTotal(unsafeRemove(p)))
```

Comment:
Isn't there a likelihood of that action being interrupted in `*>` right after `checkShutdownState`? Perhaps that part should be uninterruptible?

Reply:
I don’t see the problem if it was interrupted there? There’s a `onInterrupt` to clean it up. 

What I added is just a check that the queue wasn’t shut down before awaiting. If it was shut down, `checkShutdownState` will interrupt the current fiber (no endless waiting) and `onInterrupt` will remove the promise.

Reply:
Oh, right 👌